corehq/apps/locations/models.py

```diff
@@ -846,24 +867,30 @@ def set_location_type(self, location_type_name):
             raise LocationType.DoesNotExist(msg)
 
     @classmethod
-    def _migration_get_fields(cls):
+    def get_sync_fields(cls):
         return ["domain", "name", "site_code", "external_id", "metadata",
                 "is_archived", "latitude", "longitude"]
 
-    @classmethod
-    def _migration_get_sql_model_class(cls):
-        return SQLLocation
+    def sync_sql_location(self):
+        if not self._id:
+            self._id = uuid.uuid4().hex
```

Comment:
This seems like an odd place to set the UUID, what's the reason for putting it here rather than the save method?